src/Game/UI/Gumps/NetworkStatsGump.cs

```diff
@@ -0,0 +1,129 @@
+// 
+//  You should have received a copy of the GNU General Public License
+//  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#endregion
+
+using System.Text;
+
+using ClassicUO.Configuration;
+using ClassicUO.Game.Scenes;
+using ClassicUO.Game.UI.Controls;
+using ClassicUO.Input;
+using ClassicUO.Network;
+using ClassicUO.Renderer;
+
+namespace ClassicUO.Game.UI.Gumps
+{
+    internal class NetworkStatsGump : Gump
+    {
+
+        private readonly Label _label;
+        private readonly AlphaBlendControl _trans;
+
+        private bool _fullDisplayMode;
+
+        public NetworkStatsGump() : base(0, 0)
+        {
+            CanMove = true;
+            CanCloseWithEsc = false;
+            CanCloseWithRightClick = false;
+            AcceptMouseInput = true;
+            AcceptKeyboardInput = false;
+
+            _fullDisplayMode = !ProfileManager.Current.NetworkStatsMinimized;
+
+            Width = 200;
+            Height = 200;
+
+            Add(_trans = new AlphaBlendControl(.3f)
+            {
+                Width = Width, Height = Height
+            });
+
+            Add(_label = new Label("", true, 0x35, font: 1, style: FontStyle.BlackBorder)
+            {
+                X = 10, Y = 10
+            });
+
+            ControlInfo.Layer = UILayer.Over;
+
+            WantUpdateSize = false;
+        }
+
+        public bool FullDisplayMode
+        {
+            get => _fullDisplayMode;
+            set
+            {
+                _fullDisplayMode = value;
+                ProfileManager.Current.NetworkStatsMinimized = !_fullDisplayMode;
+            }
+        }
+
+        protected override bool OnMouseDoubleClick(int x, int y, MouseButtonType button)
+        {
+            if (button == MouseButtonType.Left)
+            {
+                FullDisplayMode = !FullDisplayMode;
+
+                return true;
+            }
+
+            return false;
+        }
+
+        public override void Update(double totalMS, double frameMS)
+        {
+            _trans.Width = Width = _label.Width + 20;
+            _trans.Height = Height = _label.Height + 20;
+
+            base.Update(totalMS, frameMS);
+        }
+
+        public override bool Draw(UltimaBatcher2D batcher, int x, int y)
+        {
+
+            if (!NetClient.Socket.IsConnected)
+            {
+                _label.Hue = NetClient.LoginSocket.Statistics.RenderedText.Hue;
+                _label.Text = NetClient.LoginSocket.Statistics.RenderedText.Text;
+            }
+            else if (!NetClient.Socket.IsDisposed)
+            {
+                _label.Hue = NetClient.Socket.Statistics.RenderedText.Hue;
+                _label.Text = NetClient.Socket.Statistics.RenderedText.Text;
+            }
+            else
+            {
+                _label.Hue = 0x20;
+                _label.Text = "Disconnected";
+            }
+                       
+            if (!FullDisplayMode) _label.Text = _label.Text.Split('\n')[0];
```

Comment:
This is cause a lot of allocation in runtime. 
What is the purpose?

Reply:
This is so the gump can be minimised with double click and show only ping value, without bytes in/out. So RenderedText gets trimmed right before the \n character that separates ping value and the rest. But can get rid of the minimised mode if it creates a lot of overhead